### DISCUSSION TITLE
Is it possible to change options based on file type?

### DISCUSSION THREAD

**Discussion by SirenGlitch** [NvChad/NvChad#2679] [Q&A] [answered]
I'm trying to set up my `custom.init` so that `vim.opt.colorcolumn = "79"` but only when the file type is Python (or when the file type is in a list if possible), whereas it would be `"80"` for any other file type. For example:

```lua
if fileType = "Python" { vim.opt.colorcolumn = "79" }

else { vim.opt.colorcolumn = "80" }
-- I apologise if the syntax is bad, I usually code in Python and have only just started learning Lua
```

**Comment by siduck** [comment, marked as answer]
just use the ftplugin feature

**Reply by SirenGlitch** [reply]
Is there a way of undoing the change after switching back from the file type? For example, if I had a buffer open for my `Dockerfile`, and one for `main.py`, would there be a way of making the `colorcolumn` change back and forth between the buffers, sort of like my "if-else" analogy earlier?

**Reply by siduck** [reply]
@SirenGlitch make an autocmd that excludes python , also ask at r/neovim, this is a neovim query, you'll get more help therre
